storage/storage.c

```diff
@@ -642,9 +642,7 @@ static void init_wiped_storage(void) {
   ui_rem = ui_total;
   ui_message = PROCESSING_MSG;
   ensure(set_pin(PIN_EMPTY, NULL), "init_pin failed");
-  if (unlocked != sectrue) {
-    memzero(cached_keys, sizeof(cached_keys));
-  }
+  unlocked = sectrue;
```

Comment:
Just a technicality: I would say that logically the storage is unlocked as soon as `cached_keys` are generated, so I recommend moving this statement to line 632 just after the `random_buffer()` call. Right now this shouldn't make any difference, but it might help us to avoid some pointless debugging in the future.

Reply:
Good point. Fixed.